MdeModulePkg/Universal/SetupBrowserDxe/Expression.c

```diff
@@ -1377,10 +1377,20 @@ IfrCatenate (
     ASSERT (Result->Buffer != NULL);
 
     TmpBuf = GetBufferForValue (&Value[0]);
-    ASSERT (TmpBuf != NULL);
+    // MU_CHANGE [BEGIN] - CodeQL change
+    if (TmpBuf == NULL) {

```

Comment:
It looks like a failure here will return with `EFI_OUT_OF_RESOURCES` and `Result` set to an empty buffer. After the `Done` label, the `Value[].Buffer` frees are in response to the `PopExpression (&Value[]);` calls. Since within that call, `PopStack()` has already moved the stack pointer, it seems a failure here should execute through the `Done` label to go ahead and free those buffers to prevent them from dangling.

Also, I only see this function `IfrCatenate()` called in one place. It might be better to update the function signature to `Result` is `NULL` on failure and just free that empty buffer to prevent it from leaking to the caller as well.

Reply:
Updated.